bpmn-model/src/main/java/io/camunda/zeebe/model/bpmn/util/time/corn/BitsCronField.java

```diff
@@ -0,0 +1,280 @@
+/*
+ * Copyright © 2017 camunda services GmbH (info@camunda.com)
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.camunda.zeebe.model.bpmn.util.time.corn;
+
+import java.time.DateTimeException;
+import java.time.temporal.Temporal;
+import java.time.temporal.ValueRange;
+
+/**
+ * Efficient bitwise-operator extension of {@link CronField}. Created using the {@code parse*}
+ * methods.
+ *
+ * @author Arjen Poutsma
```

Comment:
❌ This code seems to have been copied from somewhere and then had our license header added to it. I don't think we're legally allowed to accept this. 

Please make sure, that all code is either written by you or that it's copied with its license.

Reply:
Ah that makes sense. Would it be an idea to look for a library that can parse CRON expressions for us?

Reply:
**Quartz**? It's too heavy for us.
In addition, I have tried this [project](https://github.com/frode-carlsen/cron) before, but the support for cron is not perfect enough, so in the end, I chose to copy the implementation in [spring](https://github.com/spring-projects/spring-framework/blob/main/spring-context/src/main/java/org/springframework/scheduling/support/CronExpression.java) to deal with the problems we encountered.

Reply:
Thanks for explaining. That should be fine then 👍

Reply:
As I mentioned above, when the license is automatically replaced after compilation, I don't think about how to deal with it at present. Do you have any other suggestions? 🙇

Reply:
You'll have to configure the license plugin to leave existing licenses in place. [We've also done this in atomix](https://github.com/camunda/zeebe/blob/main/atomix/pom.xml#L56-L67). Does that help?

🤔 It makes me wonder about splitting the cron code into a new maven module, but I think we can postpone that decision.